ProjFS.Mac/PrjFSKextLogDaemon/PrjFSKextLogDaemon.cpp

```diff
@@ -21,6 +24,9 @@ using std::ostringstream;
 using std::string;
 
 static const char PrjFSKextLogDaemon_OSLogSubsystem[] = "org.vfsforgit.prjfs.PrjFSKextLogDaemon";
+static const char PanicLogDirectory[] = "/Library/Logs/DiagnosticReports";
+static const char PanicLogTimestampDirectory[] = "/Library/Logs/DiagnosticReports/VFS";
```

Comment:
I have to admit I don't know whether `/Library/Logs/DiagnosticReports` is considered a place where 3rd party software is supposed to place state. Did you look into this at all? I could imagine Apple might have designated it as a location the OS is free to delete at will - admittedly this shouldn't have any adverse side effects for this particular purpose.

Reply:
> I don't know whether /Library/Logs/DiagnosticReports is considered a place where 3rd party software is supposed to place state.

I agree, we should track state inside the directories that our applications own.

Reply:
@wilbaker do you have a good suggestion for a location here?  Happy to move it.

Reply:
How about a new "diagnostics" folder here:

`/usr/local/vfsforgit/diagnostics`

cc: @nickgra, @alameenshah for their thoughts as well

Reply:
I like that location, we should always have permissions to write to it.

Reply:
Just to check, this dir looks like we have to be 'sudo' to create. I think that's fine as long as PrjFSKextDaemon runs as root by default.

Reply:
> as long as PrjFSKextDaemon runs as root by default

I think that should be the case, cc: @alameenshah to confirm.

Reply:
@alameenshah and I looked together and it does run as root.  Yay :-)

@wilbaker this is ready for another look if you don't mind